modules/objdetect/src/qrcode.cpp

```diff
@@ -2549,23 +2584,84 @@ bool QRDecode::versionDefinition()
     }
 
     if (useCode) {
-        CV_LOG_INFO(NULL, "Version type: useCode");
+        CV_LOG_DEBUG(NULL, "Version type: useCode");
         version = (uint8_t)versionByCode;
     }
     else if (useFinderPattern ) {
-        CV_LOG_INFO(NULL, "Version type: useFinderPattern");
+        CV_LOG_DEBUG(NULL, "Version type: useFinderPattern");
         version = (uint8_t)cvRound(versionByFinderPattern);
     }
     else {
-        CV_LOG_INFO(NULL, "Version type: useTransition");
+        CV_LOG_DEBUG(NULL, "Version type: useTransition");
         version = (uint8_t)versionByTransition;
     }
     version_size = 21 + (version - 1) * 4;
     if ( !(0 < version && version <= 40) ) { return false; }
-    CV_LOG_INFO(NULL, "QR version: " << (int)version);
+    CV_LOG_DEBUG(NULL, "QR version: " << (int)version);
     return true;
 }
 
+void QRDecode::detectAlignment() {
+    vector<pair<int, int>> alignmentPositions = getAlignmentCoordinates(version);
+    if (alignmentPositions.size() > 0) {
```

Comment:
`getAlignmentCoordinates()` always returns non-empty vector. At least ` {std::make_pair(coordinates.back(), coordinates.back())};` Right?

Reply:
Non-empty vector only for versions 2+
```
static inline vector<pair<int, int>> getAlignmentCoordinates(int version) {
    if (version <= 1) return {};
```